horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/services/RunService.java

```diff
@@ -46,30 +57,25 @@ public interface RunService {
     @Operation(description = "Get extended Run information by Run ID")
     @Parameters(value = {
             @Parameter(name = "id", in = ParameterIn.PATH, description = "Run ID", example = "202"),
-            @Parameter(name = "token", in = ParameterIn.QUERY, description = "Run API token", example = "094678029a2aaf9a2847502273099bb3a1b2338c2b9c618ed09aef0181666e38")
 
     })
-    RunExtended getRun(@PathParam("id") int id,
-            @QueryParam("token") String token);
+    RunExtended getRun(@PathParam("id") int id);
```

Comment:
should the endpoints that take a token be kept (deprecated) for compatibility, sending an error when a token parameter exists ?!? ... or is it fine to just drop the token altogether ??
 
the same applies to `getRunSummary`, `getMetaData`, etc.

Reply:
IIRC query params are backward compatible in the sense that users can still provide them but they won't actually work (i.e., will be ignored).

On the other this change might affect existing users that are relying on this token feature, so we should clearly state that this has been removed and we should guide them through the new API token feature (i.e., a sort of migration guide from old tokens to new API tokens).

To answer your question, IMO I don't see a real value in keeping the query param just to return an error.. I think it is much better to completely remove it and add new migration guide to help existing users.

Reply:
I think it is fine to handle this in a migration guide and drop support in the API.  This will break integrations, but as long as the braking change is documented, with clear guidance on how to migrate, I think that is fine